src/azure-cli/azure/cli/command_modules/eventgrid/_help.py

```diff
@@ -126,9 +135,595 @@
 helps['eventgrid domain update'] = """
 type: command
 short-summary: Update a domain.
+parameters:
+  - name: --inbound-ip-rules
+    short-summary: List of inbound IP rules.
+    long-summary: |
+        List of inbound IP rules specifying IP Address in CIDR notation e.g., 10.0.0.0/8 along with corresponding Action to perform based on the match or no match of the IpMask. Possible values include - Allow.
 examples:
   - name: Update the properties of an existing domain.
-    text: az eventgrid domain update -g rg1 --name domain1 --tags Dept=IT
+    text: az eventgrid domain update -g rg1 --name domain1 --sku Basic --identity noidentity --public-network-access enabled --inbound-ip-rules 10.0.0.0/8 Allow --inbound-ip-rules 10.2.0.0/8 Allow --tags Dept=IT --sku basic
+"""
+
+helps['eventgrid partner'] = """
+type: group
+short-summary: Manage partner resources.
+"""
+
+helps['eventgrid partner registration'] = """
+type: group
+short-summary: Manage partner registrations.
+"""
+
+helps['eventgrid partner registration create'] = """
+type: command
+short-summary: Create a new partner registration.
+examples:
+  - name: Create a new partner registration with basic information.
+    text: az eventgrid partner registration create -g rg1 --name partnerRegistrationName1 --partner-name Contoso --resource-type-name Accounts --authorized-subscription-ids 533ad9de-25db-46e2-b94a-d00c37cf022b 05aa2228-7d34-4635-922d-2b582c422445
+  - name: Create a new partner registration with partner information.
+    text: az eventgrid partner registration create -g rg1 --name partnerRegistrationName1 --partner-name Contoso --resource-type-name Accounts --authorized-subscription-ids 533ad9de-25db-46e2-b94a-d00c37cf022b 05aa2228-7d34-4635-922d-2b582c422445 --description ExampleDescription --display-name ExampleDisplayName1 --logo-uri \'https://www.example.com/logo.png\' --setup-uri \'https://www.example.com\' --long-description \'This is sample description}\' --customer-service-number \'+1 800 123 4567' --customer-service-extension \'125\' --customer-service-uri \'https://www.example.com/customerservice\'
+"""
+
+helps['eventgrid partner registration list'] = """
+type: command
+short-summary: List all partner registrations in specific resource group or all under the specified azure subscription.
+examples:
+  - name: List all partner registrations in the current Azure subscription.
+    text: az eventgrid partner registration list
+  - name: List all partner registrations in a resource group.
+    text: az eventgrid partner registration list -g rg1
+  - name: List all partner registrations in a resource group whose name contains the pattern "XYZ"
+    text: az eventgrid partner registration list -g rg1 --odata-query "Contains(name, 'XYZ')"
+  - name: List all partner registrations in a resource group except the partner registration with name "name1"
+    text: az eventgrid partner registration list -g rg1 --odata-query "NOT (name eq 'name1')"
```

Comment:
should `--odata-query` be `--filter`? cli already has a global parameter named --query.

Reply:
No. This is how we handle pagination and search filtering for all existing commands.